src/ddmd/mars.d

```diff
@@ -245,6 +245,29 @@ extern (C++) void genCmain(Scope* sc)
     rootHasMain = sc._module;
 }
 
+private void semantic3OnDependencies(Module m, StringTable *importArray)
```

Comment:
At the moment it is used only in this module, so I don't see why it should be moved. If someone finds it to be useful for other purposes it can be moved later on, otherwise it should stick to the place where it is used. I think that the best approach would be to define a common.d file which implements methods that are for general purpose use, but this would mean to go through the whole dmd front end and cherry pick those methods.